DiscordBot/Commands/EventCommands.cs

```diff
@@ -106,83 +106,90 @@ public async Task EditEvent(CommandContext context, InteractivityModule interact
+                $"{context.Member.Mention} - what field do you want to edit? (``name``, ``description``, ``time``)\n");
             var editField = await GetUserResponse(context, interactivity, EventFields);
             if (editField == null)
             {
                 return;
             }
 
-            await EditEventField(context, interactivity, eventKey.Value, editField, discordEmbed);
+            await EditEventField(context, interactivity, eventKey.Value, editField, eventEmbed);
         }
 
         private static async Task EditEventField(
             CommandContext context,
             InteractivityModule interactivity,
             int eventKey,
             string editField,
-            DiscordEmbedBuilder discordEmbed)
+            DiscordEmbedBuilder eventEmbed)
         {
-            string? newName = null;
-            string? newDescription = null;
-            DateTimeOffset? newTime = null;
-
             switch (editField)
             {
                 case "name":
                     await context.RespondAsync($"{context.Member.Mention} - enter the new event name.");
-                    newName = await GetUserResponse(context, interactivity);
+                    var newName = await GetUserResponse(context, interactivity);
                     if (newName == null)
                     {
                         return;
                     }
 
-                    discordEmbed.Title = newName;
+                    eventEmbed.Title = newName;
+                    await TryEditEvent(context, eventKey, eventEmbed, newName: newName);
                     break;
                 case "description":
                     await context.RespondAsync($"{context.Member.Mention} - enter the new description.");
-                    newDescription = await GetUserResponse(context, interactivity);
+                    var newDescription = await GetUserResponse(context, interactivity);
                     if (newDescription == null)
                     {
                         return;
                     }
 
-                    discordEmbed.Description = newDescription;
+                    eventEmbed.Description = newDescription;
+                    await TryEditEvent(context, eventKey, eventEmbed, newDescription: newDescription);
                     break;
                 case "time":
                     await context.RespondAsync($"{context.Member.Mention} - enter the new event time.");
-                    newTime = await GetUserTimeResponse(context, interactivity);
+                    var newTime = await GetUserTimeResponse(context, interactivity);
                     if (newTime == null)
                     {
                         return;
                     }
 
-                    discordEmbed.Timestamp = newTime.Value.DateTime;
+                    eventEmbed.Timestamp = newTime.Value.DateTime;
+                    await TryEditEvent(context, eventKey, eventEmbed, newTime: newTime?.DateTime);
```

Comment:
Do you really need the `?` in `newTime?.DateTime` given you've already checked for null above? 🙂

Reply:
The compiler doesn't work if it's not included.

Reply:
What about `newTime.Value.DateTime`, I believe that should work?